File: [6 kyu] Which are in/main.py
def in_array(array1, array2):
    result = []
    for s1 in array1:
        for s2 in array2:
            if s1 in s2:
                if s1 not in result:
                    result.append(s1)
                break

    result.sort()

    return result


def in_array(array1, array2):
    result = set()

    for s1 in array1:
        for s2 in array2:
            if s1 in s2:
                result.add(s1)
                break

    return sorted(result)


def in_array(array1, array2):
    return sorted(set(s1 for s1 in array1 for s2 in array2 if s1 in s2))


def in_array(array1, array2):
    return sorted(set(s1 for s1 in array1 if any(s1 in s2 for s2 in array2)))


def in_array(array1, array2):
    return sorted(set(s1 for s1 in array1 if s1 in " ".join(array2)))

File: [6 kyu] Which are in/main.cpp
#include <algorithm>
#include <string>
#include <vector>

class WhichAreIn {
public:
    static std::vector<std::string> inArray(std::vector<std::string> &array1, std::vector<std::string> &array2);
};

std::vector<std::string> WhichAreIn::inArray(std::vector<std::string> &array1, std::vector<std::string> &array2) {
    std::vector<std::string> result;

    for (const auto &s1 : array1) {
        for (const auto &s2 : array2) {
            if (s2.find(s1) != s2.npos) {
                result.push_back(s1);
                break;
            }
        }
    }

    std::sort(result.begin(), result.end());

    return result;
}
